remote/advancedcontainers/engine-options.md

```diff
@@ -0,0 +1,63 @@
+---
+Order: 16
+Area: advancedcontainers
+TOCTitle: Container platforms
+PageTitle: Remote - Containers engine options
+ContentId:
+MetaDescription: Remote - Containers engine options
+DateApproved: 2/4/2022
+---
+
+# Other Container CLIs and Engines
+
+Remote - Containers interacts with CLIs; it makes no assumptions about how a container engine works and does not interact with container engines or daemon directly. You may use other Docker compliant CLIs with Remote - Containers, several of which are described below. Please note they are not officially supported.
+
+## Podman
+
+[Podman](https://podman.io/) 1.9+ is mostly compatible with Docker's CLI commands and therefore generally does work if you update the `Docker Path` setting (which may be accessed through **Remote > Containers: Docker Path**) to `podman` on Linux.
+
+![Docker Path setting](images/platform-options/docker-path-setting.png)
+
+However, certain tricks like [Docker-from-Docker do not work](https://github.com/containers/libpod/issues/4056#issuecomment-535511841) due to limitations in Podman. This affects the **Remote-Containers: Try a Development Container Sample...** and **[Remote- Containers: Clone Repository in Container Volume...](/docs/remote/containers.md#quick-start-open-a-git-repository-or-github-pr-in-an-isolated-container-volume)** commands.
+
+Docker Compose is also not supported by Podman.
+
+## Docker Contexts
+
+[Docker Contexts](https://docs.docker.com/engine/context/working-with-contexts/) allow you to interact with different hosts - you can set up contexts for each host and switch between them.
+
+You create new contexts with `docker context create`. The current context can be changed using `docker context use <context>`.
+
+The [Docker extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker) comes with `docker.host` (same as `DOCKER_HOST` env variable) and `docker.context` (same as `DOCKER_CONTEXT` env variable) user settings that are also honored by the Remote-Containers extension.
+
+> **Note**: The above settings are only visible when the Docker extension is installed. Without the Docker extension, Remote-Containers will use the current context.
+
+### Colima
+
+[Colima](https://github.com/abiosoft/colima) provides container runtimes on macOS and Linux. It's recommended to use Colima v0.2.2 or later so that VS Code can properly see containers running through Colima.
+
+Update settings in the Docker extension to use Colima:
+* `"docker.host": "unix:///Users/<username>/.colima/docker.sock",`
+* `"docker.context": "colima"`
```

Comment:
I think `docker.host` only works with the default `docker.context`. 🤔

Reply:
I was using the info in https://github.com/abiosoft/colima/issues/141#issuecomment-1021831220, but there was a response this didn't work for someone.

Besides the ssh config you shared above, are there any other steps a user must take to configure Colima?

Reply:
(I just upgraded to the latest version of Colima and Lima and now the local folder mounts are writeable from within the container. So this is more recent information.)

Colima automatically sets up a `colima` context in Docker and makes it (conveniently & a bit invasively) the active context. I have used Homebrew to also install the `docker` and `docker-compose` CLIs before running `colima start` for this to work.

Using SSH to connect to the Colima VM might be less important now. I expect the filesystem performance to still be better when using a Linux filesystem in the VM though (no measurements made). Recent versions of Lima have a command that outputs an entry for the ~/.ssh/config: `limactl show-ssh --format=config colima`. The SSH port number changes when the VM restarts, so this still needs manual updates.

Reply:
Thanks for the insight. I updated the Colima section:

- Moved out of SSH header since it won't work with Remote-SSH
- Removed SSH setup info
- Explain colima context and other info you provided here